src/emqx_packet.erl

```diff
@@ -61,8 +61,10 @@ validate(?PUBLISH_PACKET(_QoS, Topic, _, Properties, _)) ->
     ((not emqx_topic:wildcard(Topic)) orelse error(topic_name_invalid))
         andalso validate_properties(?PUBLISH, Properties);
 
-validate(?CONNECT_PACKET(#mqtt_packet_connect{properties = #{'Receive-Maximum' := 0}})) ->
-    error(protocol_error);
+validate(?CONNECT_PACKET(#mqtt_packet_connect{
+                           properties = Properties
```

Comment:
the indentation looks a bit odd to me here
IMO, more commonly adopted styles are:

1.
```
#mqtt_packet_connect{properties = Properties,
                     other_field = ...
                    })) ->
```

2.
```
#mqtt_packet_connect{
    properties = Properties,
    other_field = ...
})) ->
```
It looks like you have changed it from style 1 to 2, but indentation and `}` alignments are off.